uaclient/conftest.py

```diff
@@ -77,8 +77,9 @@ def logging_sandbox():
 @pytest.fixture
 def FakeConfig(tmpdir):
     class _FakeConfig(UAConfig):
-        def __init__(self, features_override=None) -> None:
-            super().__init__({"data_dir": tmpdir.strpath})
+        def __init__(self, cfg_overrides={}, features_override=None) -> None:
```

Comment:
Why do we need this extra `cfg_overrides` param here ?

Reply:
Because otherwise, I cannot set cfg read-only properties/attributes when instantiating.
I needed to define a log_file (which is a property without a setter) for the config, so I changed FakeConfig to accept the parameter - otherwise we could change only features, with the `features_override`